repositories/asio.BUILD.bazel

```diff
@@ -5,15 +5,10 @@ load("@rules_cc//cc:defs.bzl", "cc_library")
 
 cc_library(
     name = "asio",
-    srcs = glob(["src/*.cpp"]),
-    hdrs = glob([
-        "include/**/*.hpp",
-        "include/**/*.ipp",
+    srcs = glob([
+        "**/*.hpp",
+        "**/*.ipp",
     ]),
-    defines = [
-        "ASIO_SEPARATE_COMPILATION",
```

Comment:
What's improvement here?

Reply:
It's arguably worse to have asio as header only lib actually. But that's the way it's configured in the Bazel Central Registry. So to make sure things will work the same with Bzlmod and Workspace we need to change it to be compatible as we can't change the BCR one (at least it seems like more effort)

Reply:
FWIW, you can create a new version in BCR the way you want, and if the module maintainer agrees, you can have a separate version like 1.28.1.bcr.1. That's what I did for osqp [here](https://github.com/bazelbuild/bazel-central-registry/tree/main/modules/osqp). I wanted a specific flag to be set differently, and I created a 0.6.3.bcr.1

Just saying, so you have the option to change at source.

Reply:
I'd vote we keep the impl from this repo as it's better, and upgrade the bzl one once that's fixed.

Reply:
Updated the BUILD to match https://github.com/bazelbuild/bazel-central-registry/pull/2675/files#diff-d92e7004420fb61b89e25389540f8e1014cdde782379b35bad27adbd04f9e2c9
which in turn is based on this one.